src/pages/index.js

```diff
@@ -73,25 +115,124 @@ addCardButton.addEventListener("click", () => {
   formValidators["card-add-form"].resetValidation();
 });
 
+//Updating profile information
 function handleProfileFormSubmit(inputValues) {
-  userProfileInformation.setUserInfo(inputValues);
-  profileFormPopup.close();
+  profileFormPopup.setLoading(true);
+  api
+    .updateUserInformation(inputValues)
+    .then((result) => {
+      return result.json();
+    })
+    .then((userInformation) => {
+      userProfileInformation.setUserInfo(userInformation);
+      profileFormPopup.setLoading(false);
```

Comment:
`      profileFormPopup.setLoading(false);
`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.


**Please, correct it everywhere**